addons/point_of_sale/static/src/app/barcode/barcode_reader_service.js

```diff
@@ -7,13 +7,15 @@ import { session } from "@web/session";
 import { AlertDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
 import { BarcodeParser } from "@barcodes/js/barcode_parser";
 import { GS1BarcodeError } from "@barcodes_gs1_nomenclature/js/barcode_parser";
+import { BarcodeDialog, scanBarcode } from "@web/webclient/barcode/barcode_scanner";
 
 export class BarcodeReader {
     static serviceDependencies = ["dialog", "hardware_proxy"];
-    constructor(parser, { dialog, hardware_proxy }) {
+    constructor(parser, { dialog, hardware_proxy }, env) {
```

Comment:
instead of importing `env` here, add the "notification" service in the `serviceDependencies`. Also, in this class, you already have access to `dialog` service.